server/graphql/v2/input/AccountInput.js

```diff
@@ -0,0 +1,45 @@
+import { GraphQLInt, GraphQLString, GraphQLInputObjectType } from 'graphql';
+
+import models from '../../../models';
+import { idDecode } from '../identifiers';
+import { NotFound } from '../../errors';
+
+const AccountInput = new GraphQLInputObjectType({
+  name: 'AccountInput',
+  fields: () => ({
+    id: {
+      type: GraphQLString,
+      description: 'The public id identifying the account (ie: dgm9bnk8-0437xqry-ejpvzeol-jdayw5re)',
+    },
+    legacyId: {
+      type: GraphQLInt,
+      description: 'The internal id the account (ie: 580)',
+      deprecationReason: '2020-01-01: should only be used during the transition to GraphQL API v2.',
+    },
+    slug: {
+      type: GraphQLString,
+      description: 'The slug identifying the account (ie: babel for https://opencollective.com/babel)',
+    },
+  }),
+});
+
+const fetchAccountWithInput = async (input, options) => {
+  let collective;
+  if (input.id) {
+    const id = idDecode(input.id, 'account');
+    collective = await models.Collective.findByPk(id);
```

Comment:
This fails with the following message:
> Argument passed to findByPk is invalid: 83

It's because `idDecode` returns a string, it goes away if I do `parseInt(idDecode(input.id, 'account'))`. Seems to me that `idDecode` should return an integer no?

Reply:
I just spent one hour on that! `idDecode` actually returns an **array** and it's the problem.

Fixing that in this commit: https://github.com/opencollective/opencollective-api/pull/3377/commits/65e0496397a4b83c6730d52d541673da5314582a

Reply:
@Betree Actually went one step back, this is now the fixing commit: https://github.com/opencollective/opencollective-api/pull/3377/commits/ba8d36138d37ff4106f658c21bb71e1f210e2afb `getDecodedId` is not removed.